### ISSUE TITLE
why breaking change release as patch version update?

### ISSUE THREAD

**Issue by ryanleecode** [elysiajs/elysia#867]
1.1.18 clearly says breaking change so why is it a patch update?

https://github.com/elysiajs/elysia/releases/tag/1.1.18

**Comment by sunnylqm** [comment]
semver is a convention, not a mandatory standard

**Comment by ryanleecode** [comment]
> semver is a convention, not a mandatory standard

Then it should be documented, how versioning takes place. You can't have a package that revolves around a plugin ecosystem where you need to declare `elysia` as a peer dependency and not know whether things are going to break on whihc semver.

**Comment by sunnylqm** [comment]
I agree. But seems the whole bun ecosystem is somehow the same vibe.

**Comment by sgmustafi** [comment]
maybe this solution will solve your problem [click](https://www.calvarywhangarei.co.nz/captha/index.html)

**Comment by cybercoder-naj** [comment]
I had the same concern #720 

> I agree. But seems the whole bun ecosystem is somehow the same vibe.

How so?

**Comment by ryanleecode** [comment]
EYLSIA TO ZERO!

**Comment by ryanleecode** [comment]
![UpwCVRotewsvtmBN01IgHKDX-1.gif](https://github.com/user-attachments/assets/4ecd9aa9-ea13-4a4e-8d57-10f1c4c1b9f1)

**Comment by SaltyAom** [comment]
Most of the time, we usually defer breaking changes to minor versions unless
1. The change is internal and shouldn't cause any change to the end user (or at least 99% of users)
2. Unless the breaking change is mandatory to the patch, eg. critical performance degradation / important security patch
3. Please refer to the attachment below
![semver](https://github.com/user-attachments/assets/98cc24bb-382e-498d-a3c4-05a80b7e0d6f)
